diff --git a/aten/src/ATen/native/cuda/CuFFTUtils.h b/aten/src/ATen/native/cuda/CuFFTUtils.h
index 4b02f914d7e..b637e99bd4a 100644
--- a/aten/src/ATen/native/cuda/CuFFTUtils.h
+++ b/aten/src/ATen/native/cuda/CuFFTUtils.h
@@ -38,19 +38,25 @@ static inline std::string _cudaGetErrorEnum(cufftResult error)
       return "CUFFT_INVALID_SIZE";
     case CUFFT_UNALIGNED_DATA:
       return "CUFFT_UNALIGNED_DATA";
+    #if defined(CUFFT_INCOMPLETE_PARAMETER_LIST)
     case CUFFT_INCOMPLETE_PARAMETER_LIST:
       return "CUFFT_INCOMPLETE_PARAMETER_LIST";
+    #endif
     case CUFFT_INVALID_DEVICE:
       return "CUFFT_INVALID_DEVICE";
+    #if defined(CUFFT_PARSE_ERROR)
     case CUFFT_PARSE_ERROR:
       return "CUFFT_PARSE_ERROR";
+    #endif
     case CUFFT_NO_WORKSPACE:
       return "CUFFT_NO_WORKSPACE";
     case CUFFT_NOT_IMPLEMENTED:
       return "CUFFT_NOT_IMPLEMENTED";
 #if !defined(USE_ROCM)
+    #if defined(CUFFT_LICENSE_ERROR)
     case CUFFT_LICENSE_ERROR:
       return "CUFFT_LICENSE_ERROR";
+    #endif
 #endif
     case CUFFT_NOT_SUPPORTED:
       return "CUFFT_NOT_SUPPORTED";

